### PR TITLE
Async reservation

### LINKED ISSUE #3012: [Async Salable Status] Create infrastructure for async status update
### Description (*)

1. Post message to the queue after the reservation was placed
2. Handle message by consumer and update status in the table

### PR DESCRIPTION
**Pull request by novikor**
### Description (*)

https://github.com/magento/inventory/issues/3012

See original PR: https://github.com/magento/inventory/pull/3016

### CODE REVIEW

File: InventorySales/Model/IsProductSalableCondition/IsSalableWithReservationsCondition.php [hunk 1]
```diff
@@ -86,7 +86,7 @@ public function execute(string $sku, int $stockId): bool
         $stockItemConfiguration = $this->getStockItemConfiguration->execute($sku, $stockId);
         $qtyWithReservation = $stockItemData[GetStockItemDataInterface::QUANTITY] +
             $this->getReservationsQuantity->execute($sku, $stockId);
-        return (bool)$stockItemData[GetStockItemDataInterface::IS_SALABLE] &&
-            $qtyWithReservation > $stockItemConfiguration->getMinQty();
+
+        return $qtyWithReservation > $stockItemConfiguration->getMinQty();
```
Comment (maghamed) [inline InventorySales/Model/IsProductSalableCondition/IsSalableWithReservationsCondition.php]: I think this comparison took taken into account on the index preparation side

**Review by maghamed — COMMENTED** [review]
looks good

**Comment by swnsma** [comment]
@magento run all tests

File: InventoryIndexer/Model/Queue/RecalculateIndexSalabilityStatus.php [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+    }
+
+    /**
+     * Get current index QTY value.
+     *
+     * @param string $sku
+     * @param int $stockId
+     *
+     * @return float|null
+     * @throws LocalizedException
```
Comment (swnsma) [inline InventoryIndexer/Model/Queue/RecalculateIndexSalabilityStatus.php]: Maybe we catch exception, log it and return `null`?

File: InventoryIndexer/Plugin/InventorySales/ScheduleAfterPlaceReservationsForSalesEvent.php [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+use Magento\InventoryIndexer\Model\Queue\ReservationDataFactory;
+use Magento\InventorySales\Model\PlaceReservationsForSalesEvent;
+use Magento\InventorySalesApi\Api\Data\ItemToSellInterface;
+use Magento\InventorySalesApi\Api\Data\SalesChannelInterface;
+use Magento\InventorySalesApi\Model\GetAssignedStockIdForWebsiteInterface;
+
+/**
+ * Enqueue reservations processing after appending in order to recalculate index salability status.
+ */
+class ScheduleAfterPlaceReservationsForSalesEvent
```
Comment (swnsma) [inline InventoryIndexer/Plugin/InventorySales/ScheduleAfterPlaceReservationsForSalesEvent.php]: I do not think that 'Schedule' is right word.

File: InventoryIndexer/Plugin/InventorySales/ScheduleAfterPlaceReservationsForSalesEvent.php [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+     * @param SalesChannelInterface $salesChannel
+     *
+     * @return void
+     */
+    public function afterExecute(
+        PlaceReservationsForSalesEvent $subject,
+        $result,
+        array $items,
+        SalesChannelInterface $salesChannel
+    ) {
```
Comment (swnsma) [inline InventoryIndexer/Plugin/InventorySales/ScheduleAfterPlaceReservationsForSalesEvent.php]: ): void {

**Review by swnsma — CHANGES_REQUESTED** [review]

File: InventoryIndexer/Model/Queue/RecalculateIndexSalabilityStatus.php (hunk 2 above)
Comment (novikor) [reply InventoryIndexer/Model/Queue/RecalculateIndexSalabilityStatus.php]: Maybe

**Comment by novikor** [comment]
@swnsma , please merge.

**Pull request merged**